File: store_data_at_delete-setting-case-java/src/main/java/com/example/demo/memo/MemoEntityResponse.java
package com.example.demo.memo;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;

public record MemoEntityResponse(
        Long id,
        String title,
        String content,
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") LocalDateTime createdAt,
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") LocalDateTime updatedAt
) {
    public static MemoEntityResponse fromEntity(MemoEntity entity) {
        return new MemoEntityResponse(
                entity.getId(),
                entity.getTitle(),
                entity.getContent(),
                entity.getCreatedAt(),
                entity.getUpdatedAt()
        );
    }
}

File: store_data_at_delete-setting-case/src/main/kotlin/com/example/demo/memo/MemoEntityResponse.kt
package com.example.demo.memo

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class MemoEntityResponse(
    val id: Long,
    val title: String,
    val content: String,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdAt: LocalDateTime,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val updatedAt: LocalDateTime,
)

fun MemoEntity.toResponse(): MemoEntityResponse {
    return MemoEntityResponse(
        id = id,
        title = title,
        content = content,
        createdAt = createdAt,
        updatedAt = updatedAt,
    )
}
